File: chap2/Oui.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let _ =
  let s = "un oui ou un non ce n'est pourtant pas la même chose" in
  let longueur = String.length s in
  let n = ref 0 in
  while !n <= longueur-3 &&
    ((s.[ !n] <> 'o') || (s.[ !n+1] <> 'u') || (s.[ !n+2] <> 'i')) do
    n := !n + 1;
  done;
  if !n > longueur - 3 then
    (print_string "pas de oui"; print_newline ())
  else
    (print_int !n; print_newline ())

File: chap2/Oui.py
# Ces programmes sont sous licence CeCILL-B V1.
s = "un oui ou un non ce n'est pourtant pas la même chose"
longueur = len(s)
n = 0
while (n <= longueur-3) and ((s[n] != "o") or (s[n+1] != "u") or (s[n+2] != "i")):
   n = n + 1
if n > longueur - 3:
   print("pas de oui")
else:
   print(n)
